pkgs/build-support/writers/default.nix

```diff
@@ -22,7 +22,21 @@ rec {
       inherit interpreter;
       contentPath = content;
     }) ''
-      echo "#! $interpreter" > $out
+      # On darwin a script cannot be used as an interpreter in a shebang but
+      # there doesn't seem to be a limit to the size of shebang and multiple
+      # arguments to the interpreter are allowed.
+      if [[ -n "${toString pkgs.stdenvNoCC.isDarwin}" ]] && isScript $interpreter
+      then
+        wrapperInterpreter=$(head -1 "$interpreter" | tail -c+3)
```

Comment:
what happens if the interpreter in the interpreter is a shellscript as well? Or is this a rare usecase?

Reply:
I believe that would be indeed quite rare. It's not the case for any of the script writer provided in `lib.writers`. If a user tries to pass such interpreter, we could throw an error here and they should be able to fix it.

Reply:
addressed in 6fe292d721703094893801578c08d4f0bb33c235